UI/Source/CollectionViews/ios/CollectionViewController.swift

```diff
@@ -22,6 +23,31 @@ open class CollectionViewController: UIViewController, UICollectionViewDelegate
         layout: UICollectionViewLayout,
         context: Context,
         reuseIdProvider: CollectionViewCellReuseIdProvider = DefaultCollectionViewCellReuseIdProvider()
+    ) {
+        let dataSource = CollectionViewModelDataSource(
+            model: model.map({ [$0] }),
+            modelBinder: modelBinder,
+            viewBinder: viewBinder,
+            context: context.newScope(),
+            reuseIdProvider: reuseIdProvider)
+
+        self.collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        self.collectionView.dataSource = dataSource
+
+        dataSource.collectionView = self.collectionView
+
+        self.dataSource = dataSource
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    public init(
+        model: SectionedModelCollection,
```

Comment:
since these two inits are identical except for the minor section handling it might be good to have them both be convenience initializers that call into a private one that takes a enum { SectionedModelCollection, ModelCollection }. that way common init work doesn't have to be duped.

Reply:
Yeah definitely agree there needs to be something here but I think if we create two convenience inits for the sectioned/non-sectioned modelcollection we can't call those convenience inits from a subclasses designated init function which makes them not very useful since we expect to subclass this VC and add new stored props.